cartographer_ros/cartographer_ros/map_builder_bridge.cc

```diff
@@ -294,10 +290,11 @@ visualization_msgs::MarkerArray MapBuilderBridge::GetConstraintList() {
       continue;
     }
     const auto& submap_pose = submap_it->data.pose;
-    const auto& trajectory_node_pose =
-        all_trajectory_nodes[constraint.node_id.trajectory_id]
-                            [constraint.node_id.node_index]
-                                .global_pose;
+    const auto node_it = trajectory_nodes.find(constraint.node_id);
+    if (node_it == trajectory_nodes.end()) {
```

Comment:
Should this ever happen? If not, perhaps it should be a `CHECK`?

Reply:
Since constraints, submaps and nodes are not taken under a single lock there is a possibility of missing nodes. I planning to address this in another PR though.